### PR TITLE
feat: add listing the nicommons IDs of the installed packages

### LINKED ISSUE #271: Niconi Commons ID List
## Is your feature request related to a problem? Please describe.

AviUtl, plugins, and scripts are closely related to Niconico, and some of them ask that the ID be added to the content tree.
I want to assist with that.

Recently, a feature that allows adding IDs easily was released.
So using it is better.
https://blog.nicovideo.jp/niconews/159854.html

## Describe the solution you'd like

List up Commons IDs of installed packages that can be copied.

## Additional context

We should add an element to the data structure.
AviUtl's ID is `im1696493`.

### PR DESCRIPTION
**Pull request by hal-shu-sato**
## Description

- Add listing the nicommons IDs of the installed packages.

## Related Issue

close #271

## Motivation and Context

Refer to #271

## How Has This Been Tested?

- ~~I will add nicommons ID into the data.~~
- I tested this feature using team-apm/apm-data#190.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/renderer/main/preload.ts [hunk 1]
```diff
@@ -78,7 +79,7 @@ window.addEventListener('DOMContentLoaded', async () => {
   }
 });
 
-window.addEventListener('load', () => {
+window.addEventListener('load', async () => {
```
Comment (mitosagi) [inline src/renderer/main/preload.ts:82]: Might not be necessary.

**Review by mitosagi — CHANGES_REQUESTED** [review]
Great UI, very easy to read. The code is also basically fine.

There are a few things I would check.
`modList.ts`  has an existing bug that can be fixed as follows.
```diff
async function getInfo() {
  const modFile = await ipcRenderer.invoke('exists-temp-file', 'list.json');
  if (modFile.exists) {
    try {
      return await parseJson.getMod(modFile.path);
    } catch {
      return null;
    }
  } else {
-    return null;
+   await downloadData();
+   const downloadedModFile = await ipcRenderer.invoke('exists-temp-file', 'list.json');
+   try {
+     return await parseJson.getMod(downloadedModFile.path);
+   } catch {
+     return null;
+   }
  }
}
```
If I make this modification to PR and then launch apm from the `yarn package` with a clean install, it does not seem to work. However, the one with this modification on the `main` branch does work with a clean install. I do not know why.

File: forge.config.js [hunk 2]
```diff
@@ -62,7 +62,7 @@ module.exports = {
         mainConfig: './webpack.main.config.js',
         devServer: { liveReload: false },
         devContentSecurityPolicy:
-          "default-src 'self'; script-src 'self'; connect-src 'self'; img-src 'self' data: https://twemoji.maxcdn.com/",
+          "default-src 'self'; script-src 'self'; connect-src 'self'; img-src 'self' data: https://twemoji.maxcdn.com/ *.nicovideo.jp *.nicoseiga.jp nicovideo.cdn.nimg.jp",
```
Comment (mitosagi) [inline forge.config.js:65]:
```suggestion
          "default-src 'self'; script-src 'self'; connect-src 'self'; img-src 'self' data: https://twemoji.maxcdn.com/ https://*.nicovideo.jp https://*.nicoseiga.jp https://nicovideo.cdn.nimg.jp",
```

File: src/renderer/main/index.html [hunk 3]
```diff
@@ -5,7 +5,7 @@
 		<!-- https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP -->
 		<meta
 			http-equiv="Content-Security-Policy"
-			content="default-src 'self'; script-src 'self'; connect-src 'self' http://halshusato.starfree.jp/; img-src 'self' data: https://twemoji.maxcdn.com/"
+			content="default-src 'self'; script-src 'self'; connect-src 'self'; img-src 'self' data: https://twemoji.maxcdn.com/ *.nicovideo.jp *.nicoseiga.jp nicovideo.cdn.nimg.jp"
```
Comment (mitosagi) [inline src/renderer/main/index.html:8]:
```suggestion
			content="default-src 'self'; script-src 'self'; connect-src 'self'; img-src 'self' data: https://twemoji.maxcdn.com/ https://*.nicovideo.jp https://*.nicoseiga.jp https://nicovideo.cdn.nimg.jp"
```

File: forge.config.js (hunk 2 above)
Comment (mitosagi) [reply forge.config.js:65]: The packaged version of apm can load images with the following changes.

**Comment by mitosagi** [comment]
> Great UI, very easy to read. The code is also basically fine.
> 
> There are a few things I would check. `modList.ts` has an existing bug that can be fixed as follows.
> 
> ```diff
> async function getInfo() {
>   const modFile = await ipcRenderer.invoke('exists-temp-file', 'list.json');
>   if (modFile.exists) {
>     try {
>       return await parseJson.getMod(modFile.path);
>     } catch {
>       return null;
>     }
>   } else {
> -    return null;
> +   await downloadData();
> +   const downloadedModFile = await ipcRenderer.invoke('exists-temp-file', 'list.json');
> +   try {
> +     return await parseJson.getMod(downloadedModFile.path);
> +   } catch {
> +     return null;
> +   }
>   }
> }
> ```
> 
> If I make this modification to PR and then launch apm from the `yarn package` with a clean install, it does not seem to work. However, the one with this modification on the `main` branch does work with a clean install. I do not know why.

The cause seems to be ClipboardJS.

**Comment by hal-shu-sato** [comment]
I fixed them.
I want to merge this PR after merging #570.

**Review by mitosagi — DISMISSED** [review]
It works fine.

The ClipboardJS loading error seems to be something that existed before this PR.

**Comment by hal-shu-sato** [comment]
I rebased.

**Review by mitosagi — APPROVED** [review]
OK

**Pull request merged**